dlib/matrix/mkl_fft.h

```diff
@@ -0,0 +1,210 @@
+#ifndef DLIB_MKL_FFT_H
+#define DLIB_MKL_FFT_H
+
+#include <type_traits>
+#include <mkl_dfti.h>
+#include "fft_size.h"
+
+#define DLIB_DFTI_CHECK_STATUS(s) \
+    if((s) != 0 && !DftiErrorClass((s), DFTI_NO_ERROR)) \
+    { \
+        throw dlib::error(DftiErrorMessage((s))); \
+    }
+
+namespace dlib
+{
+    template<typename T>
+    void mkl_fft(const fft_size& dims, const std::complex<T>* in, std::complex<T>* out, bool is_inverse)
```

Comment:
Add docs for these `mkl_fft` and `mkl_fftr` functions too.

Reply:
yep